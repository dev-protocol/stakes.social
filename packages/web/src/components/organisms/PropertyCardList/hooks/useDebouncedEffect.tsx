/* eslint-disable react-hooks/exhaustive-deps */
import { useEffect } from 'react'

const useDebouncedEffect = (callback: () => any, delay = 250, dependencies?: any[]) =>
  useEffect(() => {
    const timer = setTimeout(callback, delay)

    return () => {
      clearTimeout(timer)
    }
  }, dependencies)

export default useDebouncedEffect
