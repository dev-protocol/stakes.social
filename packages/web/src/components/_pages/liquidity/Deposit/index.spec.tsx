import React from 'react'
import BigNumber from 'bignumber.js'
import { message } from 'antd'
import { act, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { allowance, balanceOf } from 'src/fixtures/_pages/liquidity/uniswap-pool/client'
import { useProvider } from 'src/fixtures/wallet/hooks'
import { Deposit } from '.'
import 'src/__mocks__/window/matchMedia.mock'

jest.mock('src/fixtures/_pages/liquidity/geyser/hooks.ts')
jest.mock('src/fixtures/_pages/liquidity/uniswap-pool/client.ts')
jest.mock('src/fixtures/_pages/liquidity/uniswap-pool/hooks.ts')
jest.mock('src/fixtures/dev-kit/hooks')
jest.mock('src/fixtures/wallet/utility')
jest.mock('src/fixtures/wallet/hooks')
jest.mock('web3')

describe(`${Deposit.name}`, () => {
  test('Snapshot', () => {
    const component = render(<Deposit geyserAddress="0xdummy" />)
    const tree = component.baseElement
    expect(tree).toMatchSnapshot()
  })
  test('react hooks: click MAX button', async () => {
    ;(balanceOf as jest.Mock).mockImplementation(() => Promise.resolve(new BigNumber('1000')))
    ;(allowance as jest.Mock).mockImplementation(() => Promise.resolve())
    ;(useProvider as jest.Mock).mockImplementation(() => ({ accountAddress: '', web3: {} }))
    message.warn = jest.fn(() => {}) as any

    const { getByText } = render(<Deposit geyserAddress="0xdummy" />)

    await act(async () => {
      await userEvent.click(getByText('Max'))
    })

    expect((message.warn as jest.Mock).mock.calls.length).toBe(0)
  })
  test('react hooks: click MAX button with balance is zero', async () => {
    ;(balanceOf as jest.Mock).mockImplementation(() => Promise.resolve(new BigNumber('0')))
    ;(allowance as jest.Mock).mockImplementation(() => Promise.resolve())
    ;(useProvider as jest.Mock).mockImplementation(() => ({ accountAddress: '', ethersProvider: {} }))
    message.warn = jest.fn(() => {}) as any

    const { getByText } = render(<Deposit geyserAddress="0xdummy" />)

    await act(async () => {
      await userEvent.click(getByText('Max'))
    })

    expect((message.warn as jest.Mock).mock.calls.length).toBe(1)
  })
  test('react hooks: click MAX button with not sign in', async () => {
    ;(useProvider as jest.Mock).mockImplementation(() => ({ accountAddress: '', web3: undefined }))
    message.warn = jest.fn(() => {}) as any

    const { getByText } = render(<Deposit geyserAddress="0xdummy" />)

    await act(async () => {
      await userEvent.click(getByText('Max'))
    })

    expect((message.warn as jest.Mock).mock.calls.length).toBe(1)
  })
})
