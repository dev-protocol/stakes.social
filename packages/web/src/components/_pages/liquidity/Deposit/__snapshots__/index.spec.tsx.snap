// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deposit Snapshot 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-vertical sc-eCImPb bZriFc"
    >
      <div
        class="ant-statistic"
      >
        <div
          class="ant-statistic-title"
        >
          Program duration
        </div>
        <div
          class="ant-statistic-content"
        >
          <span
            class="ant-statistic-content-value"
          >
            0 day(s) 00:00:00 left
          </span>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="sc-bdvvtL lmxjND"
              >
                <label
                  for="amount"
                >
                  Your amount
                </label>
                <span
                  class="ant-input-affix-wrapper ant-input-affix-wrapper-lg sc-gsDKAQ dSVFwg"
                >
                  <input
                    class="ant-input ant-input-lg"
                    id="amount"
                    type="number"
                    value=""
                  />
                  <span
                    class="ant-input-suffix"
                  >
                    <span
                      class="sc-hKwDye hiokQn"
                    >
                      ETHDEV-V2
                    </span>
                    <button
                      class="ant-btn ant-btn-primary ant-btn-sm sc-dkPtRN leCUQt"
                      type="button"
                    >
                      <span>
                        Max
                      </span>
                    </button>
                  </span>
                </span>
                <a
                  href="https://app.uniswap.org/#/add/v2/0x5cAf454Ba92e6F2c929DF14667Ee360eD9fD5b26/ETH"
                  rel="noreferrer"
                  target="_blank"
                >
                  Provide liquidity on Uniswap to get ETHDEV-V2 â†—
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="sc-bdvvtL lmxjND"
              >
                <label
                  for="estimated"
                >
                  Your estimated reward
                </label>
                <input
                  class="ant-input ant-input-lg ant-input-disabled sc-gsDKAQ dSVFwg"
                  disabled=""
                  id="estimated"
                  type="text"
                  value="0 / month"
                />
                <small>
                  *Estimated rewards assume you have achieved the maximum reward multiplier.
                </small>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-steps ant-steps-vertical ant-steps-dot"
              >
                <div
                  class="ant-steps-item ant-steps-item-process ant-steps-item-active"
                >
                  <div
                    class="ant-steps-item-container"
                  >
                    <div
                      class="ant-steps-item-tail"
                    />
                    <div
                      class="ant-steps-item-icon"
                    >
                      <span
                        class="ant-steps-icon"
                      >
                        <span
                          class="ant-steps-icon-dot"
                        />
                      </span>
                    </div>
                    <div
                      class="ant-steps-item-content"
                    >
                      <div
                        class="ant-steps-item-title"
                      >
                        Approve
                      </div>
                      <div
                        class="ant-steps-item-description"
                      >
                        <p>
                          Please approve the token transfer to deposit ETHDEV-V2.
                        </p>
                        <button
                          class="ant-btn ant-btn-primary ant-btn-lg"
                          type="button"
                        >
                          <span>
                            Approve
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-steps-item ant-steps-item-wait"
                >
                  <div
                    class="ant-steps-item-container"
                  >
                    <div
                      class="ant-steps-item-tail"
                    />
                    <div
                      class="ant-steps-item-icon"
                    >
                      <span
                        class="ant-steps-icon"
                      >
                        <span
                          class="ant-steps-icon-dot"
                        />
                      </span>
                    </div>
                    <div
                      class="ant-steps-item-content"
                    >
                      <div
                        class="ant-steps-item-title"
                      >
                        Deposit
                      </div>
                      <div
                        class="ant-steps-item-description"
                      >
                        <p>
                          Deposit ETHDEV-V2.
                        </p>
                        <button
                          class="ant-btn ant-btn-primary ant-btn-lg"
                          disabled=""
                          type="button"
                        >
                          <span>
                            Deposit
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</body>
`;
